analyzers/src/SonarAnalyzer.CSharp/Rules/CallToAsyncMethodShouldNotBeBlocking.cs

```diff
@@ -185,12 +185,17 @@ expression switch
 
         private static bool IsChainedAfterThreadPoolCall(MemberAccessExpressionSyntax memberAccess, SemanticModel semanticModel) =>
             memberAccess.Expression.DescendantNodes()
-                .OfType<MemberAccessExpressionSyntax>()
-                .Any(subMemberAccess => IsNamedSymbolOfExpectedType(subMemberAccess, semanticModel, TaskThreadPoolCalls));
+                                   .OfType<MemberAccessExpressionSyntax>()
+                                   .Any(subMemberAccess => IsNamedSymbolOfExpectedType(subMemberAccess, semanticModel, TaskThreadPoolCalls));
 
         private static bool IsNamedSymbolOfExpectedType(MemberAccessExpressionSyntax memberAccess, SemanticModel semanticModel, Dictionary<string, KnownType> expectedTypes) =>
             expectedTypes.Keys.Any(memberAccess.NameIs)
             && semanticModel.GetSymbolInfo(memberAccess).Symbol?.ContainingType?.ConstructedFrom is { } memberAccessSymbol
             && memberAccessSymbol.Is(expectedTypes[memberAccess.Name.Identifier.ValueText]);
+
+        private static bool IsInTopLevelStatement(SyntaxNodeAnalysisContext context, MemberAccessExpressionSyntax memberAccess) =>
+            context.ContainingSymbol is IMethodSymbol containingMethodSymbol
+            && containingMethodSymbol.IsMainMethod()
+            && !memberAccess.Ancestors().Any(x => x.IsKind(SyntaxKindEx.LocalFunctionStatement));
```

Comment:
According to [docs](https://rules.sonarsource.com/csharp/RSPEC-4462): 
> Main methods of Console Applications are not subject to this deadlock issue and so are ignored by this rule.

I think we should skip also the local functions in this case since they can be used only from the `Main` method